main.tf

```diff
@@ -1,6 +1,11 @@
 locals {
   // Get distinct list of domains and SANs
-  distinct_domain_names = distinct(concat([var.domain_name], data.template_file.breakup_san.*.rendered))
+  distinct_domain_names = distinct(
+    concat(
+      [var.domain_name],
+      split(":", replace(join(":", var.subject_alternative_names), "*.", ""))
```

Comment:
I think this should be this instead:
```
distinct(
  split(":", replace(join(":", concat(var.domain_name, var.subject_alternative_names)), "*.", ""))
```
Because var.domain_name might also be a wildcard, so that needs to be included in the whole `*.` filtering.